atlite/convert.py

```diff
@@ -653,6 +654,67 @@ def wind(
     )
 
 
+# #wave
+def convert_wave(ds, wec_type):
+    power_matrix = pd.DataFrame.from_dict(wec_type["Power_Matrix"])
+
+    max_pow = power_matrix.to_numpy().max()
+
+    Hs = np.ceil(ds["wave_height"] * 2) / 2
+    Tp = np.ceil(ds["wave_period"] * 2) / 2
+
+    Hs_list = Hs.to_numpy().flatten().tolist()
+    Tp_list = Tp.to_numpy().flatten().tolist()
+
+    # empty list for result
+    power_list = []
+    cases = len(Hs_list)
+    count = 0
+
+    # for loop to loop through Hs and Tp pairs and get the power output and capacity factor
+    for Hs_ind, Tp_ind in zip(Hs_list, Tp_list):
+        if count % 1000000 == 0:
+            print(f"Case {count} of {cases}: {count / cases * 100} %")
+
+        if np.isnan(Hs_ind) or np.isnan(Tp_ind):
+            power_list.append(0)
+        elif Hs_ind > 10 or Tp_ind > 18:
+            power_list.append(0)
+        else:
+            generated_power = power_matrix.loc[Hs_ind, Tp_ind]
+            power_list.append(generated_power / max_pow)
+        count += 1
+
+    # results list to numpy array
+    power_list_np = np.array(power_list)
+
+    power_list_np = power_list_np.reshape(Hs.shape)
+
+    da = xr.DataArray(
+        power_list_np, coords=Hs.coords, dims=Hs.dims, name="Power generated"
+    )
+    da.attrs["units"] = "kWh/kWp"
+    da = da.rename("specific generation")
+    da = da.fillna(0)
+
+    return da
+
+
+def wave(cutout, wec_type, **params):
+    """
+    Generate wave generation time series
+
+    evaluates the significant wave height (Hs) and wave peak period (Tp)
```

Comment:
Since `Hs` and `Tp` are MREL-specific and wouldn't make sense if using ERA5 data, it might make more sense to not reference them by these acronyms in this file.